src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -194,13 +203,13 @@ private void RequestFlush()
 
         private async Task FlushBuffersTaskLoopAsync()
         {
+            Task[] whenAnyArray = new Task[3];
```

Comment:
I don't understand the benefit of this one, since you don't reuse the array

Reply:
well is inside a while(true) and every `Task.WhenAny()` will create 2 Arrays, the one from the `params` and the  internal defensive copy. The benefit is removing 1 allocation in each loop (that's why is declared outside the loop).

Reply:
Right, I forgot about the loop